code/modules/cargo/random_stock/t1_common.dm

```diff
@@ -548,38 +527,24 @@ STOCK_ITEM_COMMON(gasmask, 2)
 	var/type = pickweight(masks)
 	new type(L)
 
-STOCK_ITEM_COMMON(cleanernades, 1.5)
-	new /obj/item/grenade/chem_grenade/cleaner(L)
-	new /obj/item/grenade/chem_grenade/cleaner(L)
-	if (prob(50))
-		new /obj/item/grenade/chem_grenade/cleaner(L)
-		new /obj/item/grenade/chem_grenade/cleaner(L)
-
 STOCK_ITEM_COMMON(mining, 2)
-	if (prob(50))
-		new /obj/item/shovel(L)
-	if (prob(50))
-		new /obj/item/pickaxe(L)
-	if (prob(50))
-		new /obj/item/clothing/glasses/material(L)
-	if (prob(50))
-		new /obj/item/device/flashlight/lantern(L)
-	if (prob(50))
-		new /obj/item/mining_scanner(L)
-	if (prob(25))
-		new /obj/item/storage/box/excavation(L)
+	var/list/mine_items = list(/obj/item/shovel, /obj/item/pickaxe, /obj/item/clothing/glasses/material,
+				/obj/item/device/flashlight/lantern, /obj/item/mining_scanner, /obj/item/storage/box/excavation)
+	for(var/i, i < rand(1, 3), i++)
```

Comment:
For the future, because BYOND:
`for(var/i = 0 to rand(1, 3))` is faster than for loops structured like this.

Reply:
I gotta resolve the merge conflict so i might as well go back and do this. thanks for the tip